lib/Auth/Source/Twitter.php

```diff
@@ -82,92 +71,98 @@ public function __construct(array $info, array $config)
      * @param array &$state  Information about the current authentication.
      */
     public function authenticate(array &$state): void
+    {
+        $this->temporaryCredentials($state);
+    }
+
+
+    /**
+     * Retrieve temporary credentials
+     *
+     * @param array &$state  Information about the current authentication.
+     */
+    private function temporaryCredentials(array &$state): void
     {
         // We are going to need the authId in order to retrieve this authentication source later
         $state[self::AUTHID] = $this->authId;
 
-        $stateID = Auth\State::saveState($state, self::STAGE_INIT);
+        $stateId = base64_encode(Auth\State::saveState($state, self::STAGE_TEMP));
```

Comment:
The only way I could get the callback to work was to base64 encode/decode the state